### ISSUE TITLE
`bool` is generic when TypeGuard is used

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1201]
**Describe the bug**
When using `TypeGuard` from `typing_extensions` in the same typing context as `bool`, Pyright thinks `bool` is generic and reports that it's partially unknown:

```
  7:9 - error: Type of parameter "value" is partially unknown
    Parameter type is "bool[Unknown]" (reportUnknownParameterType)
  7:25 - error: Declared return type, "bool[Unknown]", is partially unknown (reportUnknownVariableType)
```

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That it would not generate an error for `bool`.

**Screenshots or Code**
```python
# pyright: strict

from typing_extensions import TypeGuard

TypeGuard[object]

def foo(value: bool) -> bool:
    ...

```

**VS Code extension or command-line**
Either, 1.1.88

**Comment by layday** [comment]
Or more simply:

```py
# pyright: strict

from typing_extensions import TypeGuard

def is_str(value: object) -> TypeGuard[str]:
    # Return type, "bool[Unknown]", is partially unknown
    return isinstance(value, str)

```

**Comment by erictraut** [comment]
Apologies, this was a regression introduced with a recent change. None of the unit tests for TypeGuard have strict type checking enabled, so the regression went uncaught. It will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.89, which I just published. It will also appear in the next Pylance release.
